documentation/adr/0003-api-language.md

```diff
@@ -0,0 +1,99 @@
+- **Deciders:** Lucas Brown, Aaron Couch, Sammy Steiner, Daphne Gold
+
+## Context and Problem Statement
+
+This ADR is to decide what programming language the API server for beta.grants.gov will use. This API server will initially be responsible for responding to queries to search a replicated database, but will grow to encompass all the backend responsibilities of grants.gov
+
+In addition to the Nava and HHS teams ability to develop and maintain an API in the chosen language, it is also important that this language lends itself to open source engagement.
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+- Nava and HHS familiarity with language and common libraries
+- Performance of language and associated frameworks at scale
+- Support of Open Source community
+- Cost of necessary licenses, support, etc.
+
+## Options Considered
+
+- Python
+- Javascript and Node
+- Java
+
+
+## Decision Outcome <!-- REQUIRED -->
+Pending Discussion and Decision
+
+<!-- Chosen option: "{option 1}", because {justification. e.g., only option which meets a key decision driver | which satisfies x condition | ... }. -->
+
+### Positive Consequences <!-- OPTIONAL -->
+Pending Discussion and Decision
+
+<!-- 
+- {e.g., improved performance on quality metric, new capability enabled, ...}
+- ... 
+-->
+
+### Negative Consequences <!-- OPTIONAL -->
+Pending Discussion and Decision
+
+<!-- 
+- {e.g., decreased performance on quality metric, risk, follow-up decisions required, ...}
+- ...
+-->
+
+
+## Pros and Cons of the Options <!-- OPTIONAL -->
+
+
+### Python
+
+Python as a scripting language is fast and productive. Web-based applications can be created quickly and the code is highly readable. Python syntax is easy to understand, well-defined and extensible. APIs built over python are highly scalable and have an unmatched speed.
+
+- **Pros**
+  - Nava and HHS experience with Python and APIs built with Python
+  - Number of available libraries and frameworks
+  - Free and open source to use
+  - Ideal for data computation and complex database queries
+- **Cons**
+  - Single threaded and can be slower than other options
+
+
+### Javascript and Node
+
+Node JS is used to build fast, highly scalable network applications based on an event-driven non-blocking input/output model, single-threaded asynchronous programming. It also suits well for microservice-based products. Such Node.js utility is accomplished by such characteristics as support of a non-blocking input-output and a significant number of connections.
+
+- **Pros**
+  - Nava experience with Javascript language and Node based API servers
+  - Fast single threaded but asynchronous event driven operation
+  - Free and open source to use
+  - Same language as most front-end apps requiring less context switching for developers
+  - Could use the same framework for frontend, simplifying infrastructure
+- **Cons**
+  - Not as mature as other options
+  - Responsiveness may be slowed by any CPU intensive computation
```

Comment:
I wonder if we went with a full-stack JavaScript implementation, if we could extrapolate data-intensive and computation heavy workloads into Python scripts anyway. Or maybe in that case just better to go with full-stack Python if we have a lot of use-case for that.

Reply:
Just saying, maybe there's a world where we pick both? :D